src/MessageHandler.ts

```diff
@@ -1,19 +1,17 @@
-import config from '../config/config.json';
-
 import Discord from 'discord.js';
 import './discord/Message';
 
+import QueueItem from './queue/QueueItem';
+import SoundQueue from './queue/SoundQueue';
 import Util from './Util';
 
 export default class MessageHandler {
   private prefix: string;
-  private speaking: boolean;
-  private queue: Array<{ name: string, channel: Discord.VoiceChannel, message: Discord.Message }>;
+  private queue: SoundQueue;
 
   constructor(prefix: string) {
     this.prefix = prefix;
-    this.speaking = false;
-    this.queue = [];
+    this.queue = new SoundQueue();
```

Comment:
Try to adhere to the inversion of control design principle, that is, do not create a new queue object here, instead have it as a constructor argument.

If you think building the instance tree will be to cumbersome, at least add a constructor `constructor(queue: SoundQueue, prefix: string)` and call that from this constructor.

This will also make testing easier, because you can mock the `SoundQueue` instance.